main.c

```diff
@@ -29,21 +29,27 @@
 #define NETWORK_PREFIX_MIN (1)
 // >=26 is not supported because the recommended guest IP is set to network addr + 100 .
 #define NETWORK_PREFIX_MAX (25)
+#define NS_NAME_MAX (32)
 
-static int nsenter(pid_t target_pid)
+static int nsenter(pid_t target_pid, char *netns, char *userns)
 {
-	char userns[32], netns[32];
 	int usernsfd, netnsfd;
-	snprintf(userns, sizeof(userns), "/proc/%d/ns/user", target_pid);
-	snprintf(netns, sizeof(netns), "/proc/%d/ns/net", target_pid);
-	if ((usernsfd = open(userns, O_RDONLY)) < 0) {
-		perror(userns);
-		return usernsfd;
+	if (!netns) {
+		netns = realloc(netns, NS_NAME_MAX);
+		snprintf(netns, NS_NAME_MAX, "/proc/%d/ns/net", target_pid);
+	}
+	if (!userns) {
+		userns = realloc(userns, NS_NAME_MAX);
+		snprintf(userns, NS_NAME_MAX, "/proc/%d/ns/user", target_pid);
```

Comment:
same here

Reply:
also, why do we need to allocate them both if we will need just one?

Reply:
What do you mean by “need just one”? I may be mistaken, but are both user and network namespaces needed?